src/App.js

```diff
@@ -2,355 +2,76 @@ import React from "react";
 import "./App.css";
 import ResultComponent from "./Components/ResultComponent";
 import KeyPadComponent from "./Components/KeyPadComponent";
-import { evaluate, round } from "mathjs";
+import pressedAnotherOperator from "./pressedAnotherOperator";
+import runEquation from "./runEquation";
+import isDisplayable from "./isDisplayable";
+import cleanupEquation from "./cleanupEquation";
 
 class App extends React.Component {
   state = {
     result: "",
-    counter: 0,
-    done: false,
-    operate: undefined,
-    originalLastNum: "",
+    lastEquation: "",
   };
 
   onClick = (button) => {
-    if (button.type === "key" || button.key === "(" || button.key === ")") {
-      if (this.state.done) {
-        this.setState({
-          done: false,
-          originalLastNum: "",
-          result: button.key + "",
-        });
-      } else {
-        if (button.key === "(" || button.key === ")") {
-          this.setState({
-            result: this.state.result + button.key,
-          });
-        } else {
-          if (
-            button.key === "." ||
-            (this.state.originalLastNum + "").includes(".")
-          ) {
-            this.setState({
-              //originalLastNum: this.state.originalLastNum + button.key,
-              originalLastNum: this.recordLastNum(
-                this.state.result + button.key
-              ),
-              result: this.state.result + button.key,
-            });
-          } else {
-            this.setState({
-              //originalLastNum: button.key,
-              originalLastNum: this.recordLastNum(
-                this.state.result + button.key
-              ),
-              result: this.state.result + button.key,
-            });
-          }
-        }
-      }
-    } else {
-      if (
-        button.key !== "CE" &&
-        button.key !== "=" &&
-        button.key !== "(" &&
-        button.key !== "AC" &&
-        button.key !== ")"
-      ) {
-        this.setState({
-          operate: button.key,
-          //result: this.state.result + button.key,
-          result: this.changeKeys(this.state.result, button.key),
-        });
-      }
-    }
-    if (button.key === "=") {
-      this.calculate();
-    } else if (button.key === "AC") {
-      this.reset();
-    } else if (button.key === "CE") {
-      this.backspace();
-    }
-  };
+    const { result, lastEquation } = this.state;
 
-  recordLastNum = (result) => {
-    const regex = /[-+/*]{0,}[0-9]{1,}[.]{0,1}[0-9]*/g;
-    let results = result.match(regex) || [];
-    if (results.length === 0) {
-      return "";
-    }
-    //We need to take the last one
-    let lastNum = results[results.length - 1];
-    // If there is no operator -> we return the value
-    if (!"-+/*".includes(lastNum[0])) {
-      return lastNum;
+    if (button.key !== "=") {
+      this.setState({
+        lastEquation: "",
+      });
     }
```

Comment:
So I'm not sure but in experimenting with if it *is* possible to take it out, I ran into a bug that occurs when starting a new equation. I tested the different mutations of it with all the different operators and it occurs with all of them. The error message reads `TypeError: result.substr is not a function`. 

When examining the stack trace it seems as though `lastEquation` isn't actually being reset upon starting a new equation. To fix this I set the state of `lastEquation` to an empty string where the code for starting a new equation is, but it didn't fix this, and I think it's because it is the operator that needs to be targeted and reset. I believe this is why we were setting the state of `operator` in the last iteration of the code, and resetting it when a new equation began.

Not sure what to change to fix this, any thoughts?

<img width="478" alt="Screen Shot 2021-06-09 at 12 10 08 PM" src="https://user-images.githubusercontent.com/64915527/121390718-a4bbc080-c91b-11eb-834b-5787c9ce1857.png">

Reply:
@grimxreaper So it looks like we can't just remove it without some code breaking. Do we know why the code breaks and what we would have to do to fix it? If you figure it out, go ahead and add the change.

Reply:
@bryzaguy Actually this bug is present even without me removing it! I just happened to stumble upon it when I was experimenting with removing this code.

Reply:
So I fixed this bug by updating result rather than `lastEquation`, because result is what `hasLastChar` is changing using `.substr` with.

Reply:
@grimxreaper Great job tracking that down! I'm confused by your test added though. Was typing an equation then equals then starting another equation how you reproduced the bug? Based on your fix and the error, I would have guessed you only needed to type a number then an operator. Your commit message for a bug fix should lead with the fix, for example "fix result to number instead of string". Can we delete that block of code now? If not, what needs to change for that to happen? Once you figure it out add a commit!

Reply:
Yes that's how the bug was reproduced, that is the bug in fact, we do need the equals in the test because that is part of what causes the bug which is reflected in the stack trace.

Okay great, I will update the commit message.

That block of code still serves the purpose of displaying/saving what buttons have been pressed within the new equation, removing it causes a bug where it doesn't show the whole equation but only the last pressed button. I will investigate removing it.

Reply:
@bryzaguy okay it should be fixed now that I changed where `lastEquation` is updated!